HerPortal/Startup.cs

```diff
@@ -75,19 +75,26 @@ public void ConfigureServices(IServiceCollection services)
             })
             .AddCookie(options =>
             {
+                options.ExpireTimeSpan = TimeSpan.FromMinutes(15);
+                options.SlidingExpiration = true;
                 options.Cookie.HttpOnly = true;
                 options.Cookie.SecurePolicy = CookieSecurePolicy.Always;
+                options.Cookie.SameSite = SameSiteMode.Lax;
             })
             .AddOpenIdConnect(options =>
             {
-                options.ResponseType = "code";
+                options.NonceCookie.SameSite = SameSiteMode.Lax;
+                options.CorrelationCookie.SameSite = SameSiteMode.Lax;
+                
+                options.ResponseType = OpenIdConnectResponseType.Code;
                 options.MetadataAddress = configuration["Authentication:Cognito:MetadataAddress"];
                 options.ClientId = configuration["Authentication:Cognito:ClientId"];
                 options.ClientSecret = configuration["Authentication:Cognito:ClientSecret"];
                 options.Scope.Add("email");
                 options.Scope.Add("openid");
                 options.Scope.Add("profile");
-                options.SaveTokens = true;
+                options.SaveTokens = true; // Save tokens issued to encrypted cookies
+                options.UseTokenLifetime = false; // Don't override the cookie lifetime set above
```

Comment:
Does this mean we'll actually extend the lifetime of the id (and access) tokens?

Given that reducing the refresh token cookie lifetime doesn't really work, do you think it's still worth diverging from the default behaviour?

Reply:
I don't think we can affect the access token lifetime from here, it might mean that we keep it around for longer than is useful, but if the lifetime in Cognito ever changes to be longer it shouldn't catch us out with this setting